idaes/core/initialization/initializer_base.py

```diff
@@ -0,0 +1,375 @@
+            }
+            if not self.postcheck_summary["solver_status"]:
+                # Final solver call did not return optimal
+                self.status = InitializationStatus.Failed
+                raise InitializationError(
+                    f"{model.name} failed to initialize successfully: solver did not return "
+                    "optimal termination. Please check the output logs for more information."
+                )
+        else:
+            # Need to manually check initialization
+            # First, check that all non-stale Vars have values
+            uninit_vars = []
+            for v in model.component_data_objects(Var, descend_into=True):
+                if v.value is None:
+                    uninit_vars.append(v)
+            # Next check for unconverged equality constraints
+            uninit_const = large_residuals_set(model, self.config.constraint_tolerance)
+
+            self.postcheck_summary = {
+                "uninitialized_vars": uninit_vars,
+                "unconverged_constraints": uninit_const,
+            }
+
+            if len(uninit_const) > 0 or len(uninit_vars) > 0:
+                self.status = InitializationStatus.Failed
+                raise InitializationError(
+                    f"{model.name} failed to initialize successfully: uninitialized variables or "
+                    "unconverged equality constraints detected. Please check postcheck summary for more information."
+                )
+
+        self.status = InitializationStatus.Ok
+        return self.status
+
+    def _load_values_from_dict(self, model, initial_guesses):
+        """
+        Internal method to iterate through items in initial_guesses and set value if Var and not fixed.
+        """
+        for c, v in initial_guesses.items():
+            component = model.find_component(c)
+
+            if not isinstance(component, (Var, _VarData)):
+                self.status = InitializationStatus.Error
+                raise TypeError(
+                    f"Component {c} is not a Var. Initial guesses should only contain values for variables."
+                )
+            else:
+                if component.is_indexed():
+                    for i in component.values():
+                        if not i.fixed:
+                            i.set_value(v)
+                elif not component.fixed:
+                    component.set_value(v)
+
+    def get_submodel_initializer(self, submodel: Block):
+        """
+        Lookup Initializer object to use for specified sub-model.
+
+        This starts by checking the local mapping of objects and types, then falls back
+        to object.default_initializer, followed by a global default (if defined).
+
+        Args:
+            submodel: sub-model to get default initializer for
+
+        Returns:
+
+        """
+        # TODO: For MWE, return submodel - this will mean we run submodel.initialize()
+        return submodel
```

Comment:
I assume the idea is to update this to return an Initializer object rather than a block?

Reply:
Yes, this was a placeholder to get the example running (so that I didn't need to write the StateBlock initializers).